obspy/io/css/core.py

```diff
@@ -134,11 +135,25 @@ def _read_css(filename, **kwargs):
         else:
             read_fmt = np.dtype(dtype)
             fmt = read_fmt
-        with open(filename, "rb") as fh:
-            fh.seek(offset)
-            data = fh.read(read_fmt.itemsize * npts)
-            data = from_buffer(data, dtype=read_fmt)
-            data = np.require(data, dtype=fmt)
+
+        try:
+            # assumed that the waveform file is not compressed
+            fh = open(dfilename, "rb")
+        except FileNotFoundError as e:
+            # If does not find the waveform file referenced in the wfdisc,
+            # it will try to open a compressed .gz suffix file instead.
+            try:
```

Comment:
I haven't seen in the reference documentation for KB Core or [CSS 3.0](https://lanl-seismoacoustics.github.io/pisces/data/Anderson1990.pdf?raw=true) where adding ` .gz` to the `dfile` entry means that the waveform is gzip compressed, though it may be a convention.  This is normally done with the `datatype` entry in the `wfdisc` row, which determines how to interpret the bytes in the `dfile`, and gzip doesn't appear in the documentation for `datatype`.

Reply:
It is an operating practice that does not contradict official standards. Seismic records are continuously stored in 'css 3.0' format, waveform files uncompressed in the selected numeric representation (datatype 's4'). After a few days, the waveform files are gzip-compressed with a suffix '.gz' in a compression ratio of approximately 3:1. The 'fwdisc' table files remain intact, the 'dfile' link is not modified in any way. It is very practical if the software can work with such archived data directly and will take care of unzipping at the level of individual segments. This is supported by the CTBTO software - gbase library cssio, geotool. So I guess we are not the only ones who enjoy this method.